include/chatlogpiece.h

```diff
@@ -10,24 +10,26 @@ class chatlogpiece {
 public:
   chatlogpiece();
   chatlogpiece(QString p_name, QString p_showname, QString p_message,
-               QString p_action,int color);
+               QString p_action,int color, bool selfname);
   chatlogpiece(QString p_name, QString p_showname, QString p_message,
-               QString p_action, int color, QDateTime p_datetime);
+               QString p_action, int color, bool selfname, QDateTime p_datetime);
 
-  QString get_name();
-  QString get_showname();
-  QString get_message();
-  QString get_action();
-  QDateTime get_datetime();
-  QString get_datetime_as_string();
-  int get_chat_color();
+  QString get_name() { return name; };
+  QString get_showname() { return showname; };
+  QString get_message() { return message; };
+  QString get_action() { return action; };
+  bool get_selfname() { return selfname; };
+  QDateTime get_datetime() { return datetime; };
+  QString get_datetime_as_string() { return datetime.toString(); };
+  int get_chat_color() { return color; };
   QString get_full();
```

Comment:
```suggestion
  QString &get_name() const { return name; };
  QString &get_showname() const { return showname; };
  QString &get_message() const { return message; };
  QString &get_action() const { return action; };
  bool get_selfname() const { return selfname; };
  QDateTime &get_datetime() const { return datetime; };
  QString &get_datetime_as_string() const { return datetime.toString(); };
  int get_chat_color() const { return color; };
  QString &get_full() const;
```

However, getters are mostly pointless in this case unless you intend to fire events from them, which you are not. So, they are pointless. This seems to be more of a forceful way of declaring the class immutable, but you can already do that with `const` in C++ (it is strong immutability as opposed to the "fake" immutability you see in other languages).